src/huggingface_hub/repository.py

```diff
@@ -1032,48 +1032,70 @@ def git_checkout(self, revision: str, create_branch_ok: Optional[bool] = False):
                 except subprocess.CalledProcessError as exc:
                     raise EnvironmentError(exc.stderr)
 
-    def tag_exists(self, tag_name: str) -> bool:
+    def tag_exists(self, tag_name: str, remote: Optional[str] = None) -> bool:
         """
         Check if a tag exists or not
         """
-        try:
-            git_tags = subprocess.run(
-                ["git", "tag"],
-                stderr=subprocess.PIPE,
-                stdout=subprocess.PIPE,
-                check=True,
-                encoding="utf-8",
-                cwd=self.local_dir,
-            ).stdout.strip()
-        except subprocess.CalledProcessError as exc:
-            raise EnvironmentError(exc.stderr)
+        if remote:
+            try:
+                result = subprocess.run(
+                    ["git", "ls-remote", "origin", f"refs/tags/{tag_name}"],
+                    stderr=subprocess.PIPE,
+                    stdout=subprocess.PIPE,
+                    check=True,
+                    encoding="utf-8",
+                    cwd=self.local_dir,
+                ).stdout.strip()
+            except subprocess.CalledProcessError as exc:
+                raise EnvironmentError(exc.stderr)
 
-        git_tags = git_tags.split("\n")
-        return tag_name in git_tags
+            return len(result) != 0
+        else:
+            try:
+                git_tags = subprocess.run(
+                    ["git", "tag"],
+                    stderr=subprocess.PIPE,
+                    stdout=subprocess.PIPE,
+                    check=True,
+                    encoding="utf-8",
+                    cwd=self.local_dir,
+                ).stdout.strip()
+            except subprocess.CalledProcessError as exc:
+                raise EnvironmentError(exc.stderr)
 
-    def delete_tag(self, tag_name: str, remote: str = "origin") -> bool:
+            git_tags = git_tags.split("\n")
+            return tag_name in git_tags
+
+    def delete_tag(self, tag_name: str, remote: Optional[str] = None) -> bool:
```

Comment:
I chose to update the `remote` to be `None` by default, instead of the default `origin`. The idea behind that is to allow users to delete the tags locally without necessarily removing them remotely